src/structures/CommandInteraction.js

```diff
@@ -151,3 +151,13 @@ module.exports = CommandInteraction;
  * @external APIApplicationCommandOptionResolved
  * @see {@link https://discord.com/developers/docs/interactions/slash-commands#interaction-applicationcommandinteractiondataresolved}
  */
+
+/**
+ * @external APIInteractionDataResolvedChannel
+ * @see {@link https://discord.com/developers/docs/interactions/slash-commands#interaction-object-application-command-interaction-data-resolved-structure}
+ */
+
+/**
+ * @external APIInteractionDataResolvedGuildMember
+ * @see {@link https://discord.com/developers/docs/interactions/slash-commands#interaction-object-application-command-interaction-data-resolved-structure}
+ */
```

Comment:
Since both lead to the same link you should only add 1 and call it APIInteractionDataResolvedOption, and change it where these 2 appear

Reply:
I do not think this is the right approach. These types are imported from discord-api-types, and if they don't exist there, I don't think we should be creating them over here. Also doing this would imply both the return type of a channel and member would be the same, but this is not correct.

Reply:
Remember that these are not types and only serve for documentation purposes. Once someone clicks either one of those they will be sent to the same page and will need to figure out what they should be looking at. I argue some may even be confused and call these a bug but there's not much we can do. You could also try a weirder syntax used in typescript which would be something like `APIInteractionDataResolvedOption['member']` so that they know what property to look at when they open the page. No clue how the docs would interpret this however

Reply:
> Remember that these are not types and only serve for documentation purposes

These _are_ types, I'm not sure what you mean?

```TypeScript
export interface APIInteractionDataResolvedChannel extends Required<APIPartialChannel> {
  permissions: Permissions;
}

export interface APIInteractionDataResolvedGuildMember extends Omit<APIGuildMember, 'user' | 'deaf' | 'mute'> {
  permissions: Permissions;
}
```

Changing them away from these types means we wouldn't be using any types at all...

Reply:
No these are not types, these are only links in the documentation website. I didn't link the index.d.ts file

Reply:
Okay so, what you're getting at is we should combine `APIInteractionDataResolvedChannel` and `APIInteractionDataResolvedGuildMember` into `APIInteractionDataResolvedOption` and just make one `@external` reference, right?

The other option I was thinking of was just leading to the usual channel object/guild member object page since most other API-types do that.

Reply:
Yep that’s it

There are many different types of API external links, there’s no set rule for them so that’d be fine